src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -1053,6 +1068,190 @@ private void AddPrimarySidClaim(List<Claim> instanceClaims)
+                                c.Properties.Add(propertyValue, string.Empty);
+                                instanceClaims.Add(c);
+                            }
+                            break;
+
+
+                        case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_UINT64:
+                            long[] uintValues = new long[windowsClaim.ValueCount];
+                            Marshal.Copy(windowsClaim.Values.pUint64, uintValues, 0, (int)windowsClaim.ValueCount);
+
+                            for (int item = 0; item < windowsClaim.ValueCount; item++)
+                            {
+                                Claim c = new Claim(windowsClaim.Name, Convert.ToString((ulong)uintValues[item], CultureInfo.InvariantCulture), ClaimValueTypes.UInteger64, _issuerName, _issuerName, this);
+                                c.Properties.Add(propertyValue, string.Empty);
+                                instanceClaims.Add(c);
+                            }
+                            break;
+
+                        case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_BOOLEAN:
+                            long[] boolValues = new long[windowsClaim.ValueCount];
+                            Marshal.Copy(windowsClaim.Values.pUint64, boolValues, 0, (int)windowsClaim.ValueCount);
+
+                            for (int item = 0; item < windowsClaim.ValueCount; item++)
+                            {
+                                Claim c = new Claim(
+                                    windowsClaim.Name,
+                                    ((ulong)boolValues[item] == 0
+                                        ? Convert.ToString(false, CultureInfo.InvariantCulture)
+                                        : Convert.ToString(true, CultureInfo.InvariantCulture)),
+                                    ClaimValueTypes.Boolean,
+                                    _issuerName,
+                                    _issuerName,
+                                    this);
+
+                                c.Properties.Add(propertyValue, string.Empty);
+                                instanceClaims.Add(c);
+                            }
+                            break;
+
+
+                            // These claim types are defined in the structure found in winnt.h, but I haven't received confirmation (may  2011) that they are supported and are not enabled.
+
+                            //case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_FQBN:
+                            //    break;
+
+                            //case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_SID:
+                            //    break;
+
+                            //case Interop.ClaimSecurityAttributeType.CLAIM_SECURITY_ATTRIBUTE_TYPE_OCTET_STRING:
+                            //    break;
```

Comment:
Can we delete this?  In particular I don't like comments that say "I" in them... who the heck is the "I" here that did this investigation? :smile:

Reply:
While I want to keep the code matching netfx for goodness, I went ahead and deleted this commented out code stub.  I was on the fence when I copy/pasted it, and this was enough of a nudge.